pyNN/nest/standardmodels/electrodes.py

```diff
@@ -155,3 +155,119 @@ class NoisyCurrentSource(NestCurrentSource, electrodes.NoisyCurrentSource):
     )
     nest_name = 'noise_generator'
 
+
+# Creation of native_electrode_type
+# See Issue 506
+def get_defaults(model_name):
+    valid_types = (int, float, Sequence)
+    defaults = nest.GetDefaults(model_name)
+    variables = defaults.get('recordables', [])
+    ignore = ['archiver_length', 'available', 'Ca', 'capacity', 'elementsize',
+              'frozen', 'instantiations', 'local', 'model', 'needs_prelim_update',
+              'recordables', 'state', 't_spike', 'tau_minus', 'tau_minus_triplet',
+              'thread', 'vp', 'receptor_types', 'events', 'global_id',
+              'element_type', 'type', 'type_id', 'has_connections', 'n_synapses',
+              'thread_local_id', 'node_uses_wfr', 'supports_precise_spikes',
+              'synaptic_elements']
+    default_params = {}
+    default_initial_values = {}
+    for name, value in defaults.items():
+       if name in variables:
+            default_initial_values[name] = value
+    return default_params, default_initial_values
+
+def get_receptor_types(model_name):
+    return list(nest.GetDefaults(model_name).get("receptor_types", ('excitatory', 'inhibitory')))
+
+def native_electrode_type(model_name):
```

Comment:
This should not be in `nest/standardmodels/electrodes.py`, because a native model is by definition non-standard. Instead, create a new file `nest/electrodes.py` for this.